### DISCUSSION TITLE
Overall process of voting

### DISCUSSION THREAD

**Discussion by SupernaviX** [input-output-hk/ouroboros-leios#82] [General]
I'm still confused about the exact mechanics of voting. How do you decide that an EB has enough votes to get baked into an RB?

* It sounds like every vote has equal weight, because pools run VRF lotteries to decide how many times they vote. Pools with more stake get more votes, so there's no need for their individual votes to have more power.
* It doesn't sound like we require votes from a majority of pools, because we have a probabilistic target # of votes which is independent of network size.
* It also doesn't sound like we require a majority of stake to be represented in each vote; if one pool can make multiple votes, then "sum the stake of every voter" would double-count stake from the largest voters.

I can find descriptions of the security properties we want voting to hold. But what is the voting process which satisfies those properties? How can a collection of a few hundred votes indicate that a "majority of stake" agrees to include an EB? How do EBs "win" elections?

**Comment by pagio** [comment]
> I'm still confused about the exact mechanics of voting. 

There is a description of a possible voting scheme in the paper (which does not use ALBA), and there is also active development of ALBA based schemes. So all the details are not fixed yet.

>But what is the voting process which satisfies those properties? How can a collection of a few hundred votes indicate that a "majority of stake" agrees to include an EB? How do EBs "win" elections?

As I said there a number of options here, and the design has not been finalized yet. E.g. in the scheme proposed in the Leios paper a valid EB certificate needs to have at least k votes, for some fixed k. That is how you determine that an EB has won the election or equivalently that it has been certified. These k votes now may be compressed to a more compact form (let's call this the certificate). When I see an EB with a certificate, by the security properties of the voting scheme, I'm sure that a majority of the stake is supporting this EB and I can safely include it in the ledger.

**Comment by bwbush** [comment]
Closing this discussion because it is superseded by [the Votes and Certificates section of the CIP](https://github.com/input-output-hk/ouroboros-leios/tree/main/docs/cip#specification-for-votes-and-certificates) and the [cryptography benchmarks and reference implementation](https://github.com/input-output-hk/ouroboros-leios/blob/main/crypto-benchmarks.rs/ReadMe.md).
